deploy/helm-chart/kubernetes-replicator/templates/rbac.yaml

```diff
@@ -20,9 +20,17 @@ rules:
   - apiGroups: [ "" ]
     resources: [ "namespaces" ]
     verbs: [ "get", "watch", "list" ]
-  - apiGroups: [""] # "" indicates the core API group
-    resources: ["secrets", "configmaps"]
-    verbs: ["get", "watch", "list", "create", "update", "patch", "delete"]
+{{- range $index, $val := .Values.serviceAccount.privileges }}
+  - apiGroups:
+    {{- range $val.apiGroups }}
+      - {{ . | quote }}
+    {{- end }}
+    resources:
+    {{- range $val.resources }}
+      - {{ . | quote }}
+    {{- end }}
+    verbs: ["get", "watch", "list", "create", "update", "patch", "delete", "describe"]
+{{- end }}
```

Comment:
Generating code from a Go map is dangerous, because there is no guaranteed iteration order:

>When iterating over a map with a range loop, the iteration order is not specified and is not guaranteed to be the same from one iteration to the next. [[source](https://blog.golang.org/maps)]

This might result in frequent updates of this resource on subsequent deployments (with different permutations of the rule set), even if none of the values changed.

I'd suggest using a list instead of a map in the values (especially given the fact that `$index` isn't used for anything, here).